File: Binary Search/Python/LC 153 - FIND MINIMUM IN ROTATED SORTED ARRAY.py
# LEET CODE 153 - Find Minimum in Rotated Sorted Array
# QUESTION LINK - https://leetcode.com/problems/find-minimum-in-rotated-sorted-array/

class Solution:
    def findMin(self, nums):
        low, high = 0, len(nums) - 1
        while low < high:
            mid = (low + high) // 2
            if nums[mid] > nums[high]:
                low = mid + 1
            else:
                high = mid
        return nums[low]

File: Binary Search/Java/LC 153 - FIND MINIMUM IN ROTATED SORTED ARRAY.java
// LEET CODE 153 - Find Minimum in Rotated Sorted Array
// QUESTION LINK - https://leetcode.com/problems/find-minimum-in-rotated-sorted-array/

class Solution {
    public int findMin(int[] nums) {
        int low = 0, high = nums.length - 1;

        while (low < high) {
            int mid = (low + high) / 2;

            if (nums[mid] > nums[high]) {
                low = mid + 1;
            } else {
                high = mid;
            }
        }

        return nums[low];
    }
}
